### PR TITLE
[shopsys] renamed cookie article setting to user consent policy

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Cookies information was not descriptive enough and it was not unified across the codebase. Now it is renamed to better User consent policy
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://mg-cookie-consent.odin.shopsys.cloud
  - https://cz.mg-cookie-consent.odin.shopsys.cloud

### CODE REVIEW

File: UPGRADE-15.0.md [hunk 1]
```diff
@@ -854,6 +854,22 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
...
+    -   template `@ShopsysFramework/Admin/Content/Cookies/setting.html.twig` was moved to `@ShopsysFramework/Admin/Content/UserConsentPolicy/setting.html.twig`
+-   form type `Shopsys\FrameworkBundle\Form\Admin\Cookies\CookiesSettingFormType` was renamed to `Shopsys\FrameworkBundle\Form\Admin\UserConsentPolicy\UserConsentPolicySettingFormType`
+-   constant `Shopsys\FrameworkBundle\Component\Setting\Setting::COOKIES_ARTICLE_ID` was renamed to `Shopsys\FrameworkBundle\Component\Setting\Setting::USER_CONSENT_POLICY_ARTICLE_ID`
+-   class `Shopsys\FrameworkBundle\Model\Cookies\CookiesFacade` was renamed to `Shopsys\FrameworkBundle\Model\UserConsentPolicy\UserConsentPolicyFacade`
+    -   class no longer accepts `$environment` as a first argument in constructor
+    -   method `isCookiesConsentGiven()` was removed without a replacement
+    -   class is now strictly typed
+-   class `Shopsys\FrameworkBundle\Twig\CookiesExtension` was removed
+-   administrator roles `ROLE_COOKIES_FULL` and `ROLE_COOKIES_VIEW` were renamed to `ROLE_USER_CONSENT_POLICY_FULL` and `ROLE_USER_CONSENT_POLICY_VIEW`
```
Comment (vitek-rostislav) [inline UPGRADE-15.0.md:881]: there should be a migration for this change as well, right?

File: UPGRADE-15.0.md [hunk 2]
```diff
@@ -854,6 +854,22 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
...
+-   see #project-base-diff to update your project
```
Comment (vitek-rostislav) [inline UPGRADE-15.0.md:885]: should we mention the API changes as well?

`cookiesArticleQuery` was renamed to `userConsentPolicyArticleQuery` in `ArticleQuery` and the field name (is it called "field"?) was changed from `cookiesArticle` to `userConsentPolicyArticleQuery` :slightly_smiling_face:

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]

File: UPGRADE-15.0.md (hunk 1 above)
Comment (grossmannmartin) [reply UPGRADE-15.0.md:881]: Yes, yes it should. 
added in 3ccbfa66ca

File: UPGRADE-15.0.md (hunk 2 above)
Comment (grossmannmartin) [reply UPGRADE-15.0.md:885]: Good catch! thanks – d065a08fff

File: packages/framework/src/Migrations/Version20240527121326.php [hunk 3]
```diff
@@ -21,6 +21,52 @@ public function up(Schema $schema): void
...
+            SET roles = (
+                SELECT jsonb_agg(
+                    CASE 
+                        WHEN elem = :oldRole THEN :newRole 
+                        ELSE elem 
+                    END
+                )
+                FROM jsonb_array_elements_text(roles::jsonb) AS elem
+            )
+            WHERE :oldRole = ANY (SELECT jsonb_array_elements_text(roles::jsonb)::text)';
```
Comment (vitek-rostislav) [inline packages/framework/src/Migrations/Version20240527121326.php:53]: eh, one more reason why to avoid json in database columns? :smile:

**Review by vitek-rostislav — APPROVED** [review]

**Review by vitek-rostislav — APPROVED** [review]

File: UPGRADE-15.0.md [hunk 4]
```diff
@@ -1192,3 +1192,9 @@ We want to implement more usable UI design which will be better base for upcomin
 -   route /cookie-consent was renamed to /user-consent
 -   components and translations were also renamed/rewritten
 -   you should follow this naming convention in your app as well
+-   if you have any other special articles (other than t&c, privacy policy, user consent), you should correctly handle their error user codes
+-   if you want to use a different approach to non existent special articles (other than the one described in commit messages), be sure to modify the logic, as now
+    -   if special articles are not found, error is now not thrown
+    -   if t&c and privacy policy articles are not found, the text where they are used are the same, but without the link
+    -   if user consent article is not found, we do not display the footer link to the user consent update page, the consent update bar (`UserConsent.tsx`), and the consent update page throws returns 404
```
Comment (vitek-rostislav) [inline UPGRADE-15.0.md]:
```suggestion
    -   if user consent article is not found, we do not display the footer link to the user consent update page, the consent update bar (`UserConsent.tsx`), and the consent update page returns 404
```

File: project-base/storefront/components/Layout/Footer/Footer.tsx [hunk 5]
```diff
@@ -24,7 +25,11 @@ export type FooterProps =
 export const Footer: FC<FooterProps> = ({ simpleFooter, footerArticles, phone, opening }) => {
     const { t } = useTranslation();
     const { url } = useDomainConfig();
+    const [{ error: userConsentPolicyArticleUrlError }] = useUserConsentPolicyArticleUrlQuery();
     const [userConsentUrl] = getInternationalizedStaticUrls(['/user-consent'], url);
+    const couldNotFindUserConsentPolicyArticleUrl = userConsentPolicyArticleUrlError?.graphQLErrors.some(
+        (error) => error.extensions.userCode === 'article-not-found-user-consent-policy',
+    );
```
Comment (vitek-rostislav) [inline project-base/storefront/components/Layout/Footer/Footer.tsx]: the same block of code is used in `DeferredUserConsent.tsx`, I would say it is worth extracting into some helper method, what do you think?

**Review by vitek-rostislav — APPROVED** [review]

File: project-base/storefront/components/Layout/Footer/Footer.tsx (hunk 5 above)
Comment (sebaholesz) [reply project-base/storefront/components/Layout/Footer/Footer.tsx]: I extracted it, thanks for the suggestion.

File: UPGRADE-15.0.md (hunk 4 above)
Comment (sebaholesz) [reply UPGRADE-15.0.md]: Thanks. I fixed it in the commit message as well.

**Pull request merged**